src/map/clif.cpp

```diff
@@ -20300,24 +20300,21 @@ void clif_parse_sale_remove( int fd, struct map_session_data* sd ){
  */
 void clif_achievement_list_all(struct map_session_data *sd)
 {
-	int i, j, len, fd, *info;
-	uint16 count = 0;
-
 	nullpo_retv(sd);
 
 	if (!battle_config.feature_achievement) {
 		clif_messagecolor(&sd->bl,color_table[COLOR_RED],msg_txt(sd,772),false,SELF); // Achievements are disabled.
 		return;
 	}
 
-	fd = sd->fd;
-	count = sd->achievement_data.count; // All achievements should be sent to the client
-	len = (50 * count) + 22;
+	int count = sd->achievement_data.count; // All achievements should be sent to the client
+	int len = (50 * count) + 22;
 
 	if (len <= 22)
```

Comment:
Instead of the length check, check the sd count you assign above and only calculate length if there is an existent entry.